blindsign/blindrsa/common.go

```diff
@@ -0,0 +1,121 @@
+package blindrsa
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/sha512"
+	"crypto/subtle"
+	"errors"
+	"hash"
+	"io"
+	"math/big"
+
+	"github.com/cloudflare/circl/blindsign/blindrsa/internal/keys"
+)
+
+// ConvertHashFunction converts a crypto.Hash function to an equivalent hash.Hash type.
+func ConvertHashFunction(hash crypto.Hash) hash.Hash {
+	switch hash {
+	case crypto.SHA256:
+		return sha256.New()
+	case crypto.SHA384:
+		return sha512.New384()
+	case crypto.SHA512:
+		return sha512.New()
+	default:
+		panic(errUnsupportedHashFunction)
+	}
```

Comment:
Since every `crypto.Hash` has a `New` method to get a `hash.Hash`, this code can be simplified

```suggestion
	switch hash {
	case crypto.SHA256, crypto.SHA384, crypto.SHA512:
		return hash.New()
	default:
		panic(errUnsupportedHashFunction)
	}
```

Reply:
Reverting this change as it doesn't work:

```
panic: crypto: requested hash function #6 is unavailable [recovered]
	panic: crypto: requested hash function #6 is unavailable
```

Reply:
must keep the imports to the packages.
```go
import _ "crypto/sha256"
import _ "crypto/sha512"
```